membership-common/src/main/scala/com/gu/memsub/subsv2/Plan.scala

```diff
@@ -122,178 +108,123 @@ case object Years extends UpToPeriodsType {
-    model: String,
-    name: String,
-    `type`: String,
     endDateCondition: EndDateCondition,
     upToPeriods: Option[Int],
     upToPeriodsType: Option[UpToPeriodsType],
-)
-
-object ZuoraCharge {
-  def apply(
-      productRatePlanChargeId: ProductRatePlanChargeId,
-      pricing: PricingSummary,
-      billingPeriod: Option[ZBillingPeriod],
-      specificBillingPeriod: Option[Int],
-      model: String,
-      name: String,
-      `type`: String,
-      endDateCondition: EndDateCondition,
-      upToPeriods: Option[Int],
-      upToPeriodsType: Option[UpToPeriodsType],
-  ): ZuoraCharge = ZuoraCharge(
-    SubscriptionRatePlanChargeId(""),
-    productRatePlanChargeId,
-    pricing,
-    billingPeriod,
-    specificBillingPeriod,
-    model,
-    name,
-    `type`,
-    endDateCondition,
-    upToPeriods,
-    upToPeriodsType,
-  )
+) {
+
+  def billingPeriod: Validation[String, BillingPeriod] =
+    (endDateCondition, zBillingPeriod) match {
+      case (FixedPeriod, Some(ZSpecificWeeks))
+          if specificBillingPeriod.exists(numberOfWeeks => numberOfWeeks == 6 || numberOfWeeks == 7) &&
+            upToPeriods.contains(1) &&
+            upToPeriodsType.contains(BillingPeriods) =>
+        Validation.success[String, BillingPeriod](SixWeeks)
+      case (FixedPeriod, Some(zPeriod))
+          if upToPeriods.contains(1) &&
+            upToPeriodsType.contains(BillingPeriods) =>
+        zPeriod match {
+          case ZYear => Validation.success[String, BillingPeriod](OneYear)
+          case ZQuarter => Validation.success[String, BillingPeriod](ThreeMonths)
+          case ZTwoYears => Validation.success[String, BillingPeriod](TwoYears)
+          case ZThreeYears => Validation.success[String, BillingPeriod](ThreeYears)
+          case ZSemiAnnual => Validation.success[String, BillingPeriod](SixMonths)
+          case _ => Validation.f[BillingPeriod](s"zuora fixed period was $zPeriod")
+        }
+      case (SubscriptionEnd, Some(zPeriod)) =>
+        zPeriod match {
+          case ZMonth => Validation.success[String, BillingPeriod](Month)
+          case ZQuarter => Validation.success[String, BillingPeriod](Quarter)
+          case ZYear => Validation.success[String, BillingPeriod](Year)
+          case ZSemiAnnual => Validation.success[String, BillingPeriod](SixMonthsRecurring)
+          case _ => Validation.f[BillingPeriod](s"zuora recurring period was $zPeriod")
+        }
+      case (OneTime, None) => Validation.success[String, BillingPeriod](OneTimeChargeBillingPeriod) // This represents a one time rate plan charge
+      case _ =>
+        Validation.f[BillingPeriod](
+          s"period =${zBillingPeriod} specificBillingPeriod=${specificBillingPeriod} uptoPeriodsType=${upToPeriodsType}, uptoPeriods=${upToPeriods}",
+        )
+    }
+
 }
 
 /** Low level model of a rate plan, as it appears on a subscription in Zuora
   */
-case class SubscriptionZuoraPlan(
+case class RatePlan(
```

Comment:
although this is still the low level model, now it has a load of helper methods on it to mimic the fields that were on the old high level model.  The new methods take the catalog as a parameter in most cases.